features/events/traps/src/test/java/org/opennms/netmgt/trapd/TrapdListenerBlueprintIT.java

```diff
@@ -195,4 +216,104 @@ public void setTrapIdentity(TrapIdentity trapIdentity) {
+                trapReciever=new TrapReceiverImpl(getTrapConfiguration());
+                from("http://localhost:10500/trapd-configuration-withoutSnmpV3Users.html").bean(jaxbUnmarshaller).bean(TrapdConfigBean.class,
+                        "onUpdate").bean(trapReciever, "setTrapdConfig").to("mock:result");
+            }
+        });
+        camelContext.start();
+
+        mockEndPoint = camelContext.getEndpoint("mock:result", MockEndpoint.class);
+        mockEndPoint.expectedMessageCount(1);
+
+        producerTemplate = camelContext.createProducerTemplate();
+        producerTemplate.start();
+        
+        mockEndPoint.assertIsSatisfied();
+
+        producerTemplate.stop();
+        camelContext.stop();
+        
+     
+    }
+    
+    @Test
+    @JUnitHttpServer(port = 10500)
+    public void testRouteWithSnmpV3Users() throws Exception {
+        
+        camelContext = new DefaultCamelContext();
+        camelContext.addRoutes(new RouteBuilder() {
+            @Override
+            public void configure() throws Exception {
+                jaxbUnmarshaller=new JaxbUtilsUnmarshalProcessor(TrapdConfiguration.class);
+                trapReciever=new TrapReceiverImpl(getTrapConfiguration());
+                from("http://localhost:10500/trapd-configuration-withSnmpV3Users.html").bean(jaxbUnmarshaller).bean(TrapdConfigBean.class,
+                        "onUpdate").bean(trapReciever, "setTrapdConfig").to("mock:result");
+            }
+        });
+        camelContext.start();
+
+        mockEndPoint = camelContext.getEndpoint("mock:result", MockEndpoint.class);
+        mockEndPoint.expectedMessageCount(1);
+
+        producerTemplate = camelContext.createProducerTemplate();
+        producerTemplate.start();
+        
+        mockEndPoint.assertIsSatisfied();
+        
+        producerTemplate.stop();
+        camelContext.stop();
+
+    }
+    
+    @Test
+    @JUnitHttpServer(port = 10500)
+    public void testRouteWithSnmpV3UsersMultipleTimes() throws Exception {
+        camelContext = new DefaultCamelContext();
+        camelContext.addRoutes(new RouteBuilder() {
+            @Override
+            public void configure() throws Exception {
+                jaxbUnmarshaller=new JaxbUtilsUnmarshalProcessor(TrapdConfiguration.class);
+                trapReciever=new TrapReceiverImpl(getTrapConfiguration());
+                from("http://localhost:10500/trapd-configuration-withSnmpV3Users.html").bean(jaxbUnmarshaller).bean(TrapdConfigBean.class,
+                        "onUpdate").bean(trapReciever, "setTrapdConfig").to("mock:result");
+            }
+        });
+        camelContext.start();
+        
+        for (int i = 0; i < 10; i++) {
+        mockEndPoint = camelContext.getEndpoint("mock:result", MockEndpoint.class);
+        mockEndPoint.expectedMessageCount(1);
+            producerTemplate = camelContext.createProducerTemplate();
+            producerTemplate.start();
```

Comment:
I don't think that this loop is doing anything, you're just starting the producerTemplate over and over without sending any messages. Change this test to use a timer or something so that you can invoke the HTTP route multiple times.